apa.csl

```diff
@@ -155,88 +358,83 @@
   </macro>
   <macro name="access">
     <choose>
-      <if type="thesis report" match="any">
-        <choose>
-          <if variable="DOI" match="any">
-            <text variable="DOI" prefix="https://doi.org/"/>
-          </if>
-          <else-if variable="URL" match="any">
-            <group>
-              <text term="retrieved" text-case="capitalize-first" suffix=" "/>
-              <text term="from" suffix=" "/>
-              <text variable="URL"/>
-            </group>
-          </else-if>
-          <else-if variable="archive" match="any">
-            <group>
-              <text term="retrieved" text-case="capitalize-first" suffix=" "/>
-              <text term="from" suffix=" "/>
-              <text variable="archive" suffix="."/>
-              <text variable="archive_location" prefix=" (" suffix=")"/>
-            </group>
-          </else-if>
-        </choose>
+      <if variable="DOI" match="any">
+        <text variable="DOI" prefix="https://doi.org/"/>
       </if>
-      <else>
-        <choose>
-          <if variable="DOI">
-            <text variable="DOI" prefix="https://doi.org/"/>
-          </if>
-          <else>
+      <else-if variable="URL">
+        <group delimiter=" ">
+          <group delimiter=" " suffix=":">
+            <text term="retrieved" text-case="capitalize-first"/>
             <choose>
               <if type="post post-weblog webpage" match="any">
-                <group delimiter=" ">
-                  <text term="retrieved" text-case="capitalize-first" suffix=" "/>
-                  <group>
-                    <date variable="accessed" form="text" suffix=", "/>
-                  </group>
-                  <text term="from"/>
-                  <text variable="URL"/>
-                </group>
+                <date variable="accessed" form="text" suffix=","/>
               </if>
-              <else>
-                <group>
-                  <text term="retrieved" text-case="capitalize-first" suffix=" "/>
-                  <text term="from" suffix=" "/>
-                  <text variable="URL"/>
-                </group>
-              </else>
             </choose>
-          </else>
-        </choose>
+            <text term="from"/>
+            <choose>
+              <if type="report">
+                <choose>
+                  <if variable="author editor translator" match="any">
+                    <!-- This should be localized -->
+                    <text variable="publisher" suffix=" website"/>
+                  </if>
+                </choose>
+              </if>
+              <else-if type="post post-weblog webpage">
+                <!-- This should be localized -->
+                <text variable="container-title" suffix=" website"/>
+              </else-if>
+            </choose>
+          </group>
+          <text variable="URL"/>
+        </group>
+      </else-if>
+      <else-if variable="archive" type="article dataset report thesis" match="all">
```

Comment:
This has been how citeproc-js has worked for years, and several other styles use type=“aaa bbb” match=“all”. citeproc-js seems to treat multiples in type=“ “ as meaning “type is one in this list”.

Reply:
I'm pretty sure that pandoc-citeproc does this correctly, though, so just accepting this isn't really an option. If we have remaining styles with this, we'll need to fix them. That said:

> and several other styles use type=“aaa bbb” match=“all”

Do you have examples? I can't find any.

Reply:
(Just to be clear, I'm not trying to argue with you, just being clear about why I wrote it that way [because it worked in citeproc-js, there were a few examples, and it seemed possible that type="" was intended to be evaluated as "type is one of these"]. Obviously it's not a big deal to add the four lines of code if this behavior is unintended.)

The styles that still have multiple types and match="all" are:
```
harvard-university-for-the-creative-arts.csl
  221,16:       <else-if type="report entry-dictionary entry-encyclopedia motion_picture chapter speech song paper-conference article-journal book" match="all" variable="URL">

lluelles-no-ibid.csl
  732,30:           <if locator="page" type="chapter article-magazine article-newspaper" match="all">
  742,20:           <else-if type="legislation bill" locator="section" match="all">

lluelles.csl
  757,30:           <if locator="page" type="chapter article-magazine article-newspaper" match="all">
  767,20:           <else-if type="legislation bill" locator="section" match="all">
  854,46:                   <if variable="title-short" type="legislation bill entry-dictionary" match="all">

mcgill-fr.csl
  918,20:           <else-if type="legislation bill" locator="section" match="all">
  977,46:                   <if variable="title-short" type="legislation bill entry-dictionary" match="all">

universite-de-montreal-faculte-de-musique.csl
  621,20:           <else-if type="legislation bill" locator="section" match="all">
  680,46:                   <if variable="title-short" type="legislation bill entry-dictionary" match="all">

associacao-brasileira-de-normas-tecnicas-unirio-eipp.csl
  61,11:       <if match="all" type="book chapter">
  86,11:       <if match="all" variable="edition" type="book chapter">
  145,15:           <if match="all" variable="volume" type="book chapter article-journal thesis">
  161,15:           <if match="all" variable="page" type="chapter article-journal">
  169,15:           <if match="all" variable="number-of-pages" type="book thesis">

journal-of-the-warburg-and-courtauld-institutes.csl
  480,16:       <else-if match="all" type="webpage post-weblog">
```